src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceLookup/ServiceProviderEngineScope.cs

```diff
@@ -19,12 +20,13 @@ internal class ServiceProviderEngineScope : IServiceScope, IServiceProvider, IAs
         private bool _disposed;
         private readonly object _disposelock = new object();
 
-        public ServiceProviderEngineScope(ServiceProviderEngine engine)
+        public ServiceProviderEngineScope(ServiceProviderEngine engine, bool forRoot = false)
```

Comment:
nit: isRoot

Reply:
Also add a comment about the fact that we expect lots of contention for singletons, that's why we have a concurrent dictionary.